### PR TITLE
synchrony: init at 2.4.5

### LINKED ISSUE #352355: Package request: `synchrony`
**Project description**

Javascript cleaner & deobfuscator 

**Metadata**

* homepage URL: https://github.com/relative/synchrony
* source URL: https://github.com/relative/synchrony
* license: gpl3
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by pluiedev**
Fixes #352355

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sy/synchrony/package.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      --replace-fail "require('../')" "require('$out/share/synchrony')"
+    patchShebangs $out/share/synchrony/cli.js
+
+    runHook postFixup
+  '';
+
+  passthru = {
+    tests.deobfuscate =
+      let
+        obfuscated = fetchurl {
```
Comment (FliegendeWurst) [inline pkgs/by-name/sy/synchrony/package.nix:67]: Why not fetch both obfuscated and deobfuscated?

File: pkgs/by-name/sy/synchrony/package.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "Simple deobfuscator for mangled or obfuscated JavaScript files";
+    homepage = "https://deobfuscate.relative.im/";
+    license = with lib.licenses; [ gpl3Only ];
+    maintainers = with lib.maintainers; [ pluiedev ];
+    platforms = lib.platforms.all;
```
Comment (FliegendeWurst) [inline pkgs/by-name/sy/synchrony/package.nix:86]:
```suggestion
    platforms = nodejs.meta.platforms;
```

File: pkgs/by-name/sy/synchrony/package.nix (hunk 1 above)
Comment (pluiedev) [reply pkgs/by-name/sy/synchrony/package.nix:67]: There's no deobfuscated version provided by upstream — I had to run this on my local machine

Comment (FliegendeWurst) [reply pkgs/by-name/sy/synchrony/package.nix:67]: I suppose that doesn't stop you from uploading the deobfuscated file to gist.github.com though.

Comment (pluiedev) [reply pkgs/by-name/sy/synchrony/package.nix:67]: > I suppose that doesn't stop you from uploading the deobfuscated file to gist.github.com though.

True, though that feels like more work than just sourcing from a file in-tree

Comment (SuperSandro2000) [reply pkgs/by-name/sy/synchrony/package.nix:65]: I would suggest to not commit the deobfuscate version at all and just check if it works.

**Review by SuperSandro2000 — DISMISSED** [review]

**Review by getchoo — APPROVED** [review]

**Comment by getchoo** [comment]
@ofborg eval
(this is a bit old)

**Comment by Aleksanaa** [comment]
@ofborg build synchrony

**Pull request merged**